src/Component/Result/Facet/Buckets.php

```diff
@@ -18,14 +18,22 @@ class Buckets implements FacetResultInterface, \IteratorAggregate, \Countable
      */
     protected $buckets;
 
+    /**
+     * numBuckets.
+     *
+     * @var int|null
+     */
+    protected $numBuckets;
+
     /**
      * Constructor.
      *
      * @param Bucket[] $buckets
      */
-    public function __construct(array $buckets)
+    public function __construct(array $buckets, $numBuckets = null)
```

Comment:
We're down to nitpicking (which is always a good sign!): `$numBuckets` is missing an `@param` and a typehint.

Reply:
Thank you for your patience guiding me through the process. I think we're nearly there! Will make sure to re-read this PR before future submissions as it now lists all the things i forgot to do ;-)